src/Workspaces/Core/Portable/Workspace/TextDocumentEventArgs.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class TextDocumentEventArgs : EventArgs
+    {
+        public TextDocument Document { get; }
```

Comment:
Random note for the (eventual) API review, but ignore for now: I think @sharwell and I when we were doing some of the source generator stuff also regretted that this was actually a document object versus say Solution + DocumentId. This came up when we had source generated documents being open, since actually getting a document object is more expensive and potentially requires asynchrony under the covers. We were able to make it work, but maybe we can also change this if we end up deprecating the older APIs.

Reply:
Sounds reasonable!